src/classes/Warehouse.java

```diff
@@ -196,5 +201,57 @@ private void scnRemoveCart() {
 
         carts.remove(cartsName.get(intChoice));
     }
+
+    public void scnResearch() {
+        Scanner scanner = new Scanner(System.in);
+        String input;
+        int inputDigit;
+        int intChoice;
+        do {
+            System.out.println("Seleziona il criterio di ricerca (usa i numeri):\n" +
+                    "1.Produttore   2.Modello    3.ID   e. Torna alla scehrmata precedente");
+            input = scanner.nextLine();
+            if (input.equalsIgnoreCase("e")) {
+                return;
+            }
+            while (!input.matches("[1-3]")) {
+                System.out.println("Usa i numeri per effettuare la scelta o 'e' per tornare alla schermata precedente\n");
+                input = scanner.nextLine();
+            }
+            intChoice = Integer.parseInt(input);
+        } while (intChoice < 1 || intChoice > 3);
+        switch (intChoice) {
+            case 1:
+                System.out.println("Seleziona il produttore che vuoi cercare:");
+                input = scanner.nextLine();
+                for (Product item : stock.keySet()) {

```

Comment:
Questa è la parte da testare, andrebbe estratta in un altro metodo "searchByProducer" che prende in input il nome del produttore.